sti-main/src/main/java/uk/ac/shef/dcs/sti/core/algorithm/tmp/TMPOdalicInterpreter.java

```diff
@@ -71,6 +73,26 @@ public TAnnotation start(Table table, Constraints constraints) throws STIExcepti
           .compute(table, constraints.getSubjectColumnPosition(), ignoreColumnsArray);
       tableAnnotations.setSubjectColumn(subjectColumnScores.get(0).getKey());
       
+      // when the column does not contain Named entity as the most frequent data type,
+      // we will not disambiguate (and so classify) them, except for
+      // the user defined constraints
+      Set<Ambiguity> newAmbiguities = new HashSet<>(constraints.getAmbiguities());
```

Comment:
Ok, lets leave that for now, if it works correctly, that is enough for now. Performance improvements may be done later if needed.